pkg/grpcfactory/grpc_factory.go

```diff
@@ -58,25 +59,24 @@ func NewCOSIProvisionerClient(ctx context.Context, address string, dialOpts []gr
 		return nil, err
 	}
 	if addr.Scheme != "unix" {
-		err := errors.New("Address must be a unix domain socket")
-		klog.ErrorS(err, "Unsupported scheme", "expected", "unix", "found", addr.Scheme)
-		return nil, fmt.Errorf("unsupported scheme: %w", err)
+		err := fmt.Errorf("unsupported scheme: expected 'unix', found '%s'", addr.Scheme)
+		klog.ErrorS(err, "Invalid address scheme")
+		return nil, err
 	}
-	for _, interceptor := range interceptors {
-		dialOpts = append(dialOpts, grpc.WithChainUnaryInterceptor(interceptor))
+	if len(interceptors) > 0 {
+		dialOpts = append(dialOpts, grpc.WithChainUnaryInterceptor(interceptors...))
 	}
-	ctx, cancel := context.WithTimeout(ctx, maxGrpcBackoff)
-	defer cancel()
-	conn, err := grpc.DialContext(ctx, address, dialOpts...)
+	// Proceed without grpc.WithBlock(), allowing connection to establish asynchronously
+	conn, err := grpc.Dial(address, dialOpts...)
 	if err != nil {
 		klog.ErrorS(err, "Connection failed", "address", address)
 		return nil, err
 	}
 	return &COSIProvisionerClient{
 		address:           address,
 		conn:              conn,
-		identityClient:    cosi.NewIdentityClient(conn),
-		provisionerClient: cosi.NewProvisionerClient(conn),
+		IdentityClient:    cosi.NewIdentityClient(conn),
+		ProvisionerClient: cosi.NewProvisionerClient(conn),
```

Comment:
Ah I didn't know you could assign an embedded interface like this, good to know!